src/applets/clock/ClockApplet.vala

```diff
@@ -265,19 +211,42 @@ public class ClockApplet : Budgie.Applet {
 		manager.register_popover(widget, popover);
 	}
 
-	protected void on_settings_change(string key) {
-		switch (key) {
-			case "clock-format":
-				SignalHandler.block((void*)this.clock_format, this.check_id);
-				ClockFormat f = (ClockFormat)settings.get_enum(key);
-				ampm = f == ClockFormat.TWELVE;
-				clock_format.set_active(f == ClockFormat.TWENTYFOUR);
-				this.update_clock();
-				SignalHandler.unblock((void*)this.clock_format, this.check_id);
+	private void update_setting(string schema, string key) {
+		switch (schema) {
+			case CLOCK_SETTINGS_SCHEMA:
```

Comment:
Using a if / else if for schema would clean up the indentation levels a bit. A switch + case for two possible cases is a bit overkill.